packages/backend-platform/src/WebsocketService.ts

```diff
@@ -0,0 +1,1348 @@
+  // Value: InternalSubscription object with channels, callback and metadata
+  readonly #subscriptions = new Map<string, InternalSubscription>();
+
+  // Channel-based callback storage
+  // Key: channel name (serves as unique identifier)
+  // Value: ChannelCallback configuration
+  readonly #channelCallbacks = new Map<string, ChannelCallback>();
+
+  /**
+   * Creates a new WebSocket service instance
+   *
+   * @param options - Configuration options including messenger
+   */
+  constructor(
+    options: WebSocketServiceOptions & { messenger: WebSocketServiceMessenger },
+  ) {
+    this.#messenger = options.messenger;
+
+    this.#options = {
+      url: options.url,
+      timeout: options.timeout ?? 10000,
+      reconnectDelay: options.reconnectDelay ?? 500,
+      maxReconnectDelay: options.maxReconnectDelay ?? 5000,
+      requestTimeout: options.requestTimeout ?? 30000,
+    };
+
+    // Register action handlers
+    this.#messenger.registerActionHandler(
+      `BackendWebSocketService:init`,
+      this.init.bind(this),
+    );
+
+    this.#messenger.registerActionHandler(
+      `BackendWebSocketService:connect`,
+      this.connect.bind(this),
+    );
+
+    this.#messenger.registerActionHandler(
+      `BackendWebSocketService:disconnect`,
+      this.disconnect.bind(this),
+    );
+
+    this.#messenger.registerActionHandler(
+      `BackendWebSocketService:sendMessage`,
+      this.sendMessage.bind(this),
+    );
+
+    this.#messenger.registerActionHandler(
+      `BackendWebSocketService:sendRequest`,
+      this.sendRequest.bind(this),
+    );
+
+    this.#messenger.registerActionHandler(
+      `BackendWebSocketService:getConnectionInfo`,
+      this.getConnectionInfo.bind(this),
+    );
+
+    this.#messenger.registerActionHandler(
+      `BackendWebSocketService:getSubscriptionByChannel`,
+      this.getSubscriptionByChannel.bind(this),
+    );
+
+    this.#messenger.registerActionHandler(
+      `BackendWebSocketService:isChannelSubscribed`,
+      this.isChannelSubscribed.bind(this),
+    );
+
+    this.#messenger.registerActionHandler(
+      `BackendWebSocketService:addChannelCallback`,
+      this.addChannelCallback.bind(this),
+    );
+
+    this.#messenger.registerActionHandler(
+      `BackendWebSocketService:removeChannelCallback`,
+      this.removeChannelCallback.bind(this),
+    );
+
+    this.#messenger.registerActionHandler(
+      `BackendWebSocketService:getChannelCallbacks`,
+      this.getChannelCallbacks.bind(this),
+    );
+
+    this.init().catch((error) => {
+      console.error('WebSocket service initialization failed:', error);
+    });
+  }
+
+  /**
+   * Initializes and connects the WebSocket service
+   *
+   * @returns Promise that resolves when initialization is complete
+   */
+  async init(): Promise<void> {
+    try {
+      await this.connect();
+    } catch (error) {
+      const errorMessage =
+        error instanceof Error ? error.message : 'Unknown initialization error';
```

Comment:
We have a utility in `@metamask/utils` to convert a throwable to a string, what do you think about using that? 

``` suggestion
      const errorMessage = getErrorMessage(error);
```